firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteStore.java

```diff
@@ -547,7 +547,10 @@ private void raiseWatchSnapshot(SnapshotVersion snapshotVersion) {
 
     // Re-establish listens for the targets that have been invalidated by existence filter
     // mismatches.
-    for (int targetId : remoteEvent.getTargetMismatches()) {
+    for (Map.Entry<Integer, QueryPurpose> entry : remoteEvent.getTargetMismatches().entrySet()) {
+
+      int targetId = entry.getKey();
```

Comment:
nit: please remove the blank lines above and below `int targetId = entry.getKey();`